docs/config/configuring-authentication.rst

```diff
@@ -0,0 +1,171 @@
+
+.. note:: In this command ``<username>`` is a unique username for an API user.
+   Please note though that ``flocker-deploy`` requires these files to be named :file:`user.crt` and :file:`user.key`.
+   If you intend on using ``flocker-deploy``, you will need to rename your files to :file:`user.crt` and :file:`user.key`.
+
+The two files generated will correspond to the username you specified in the command, in this example :file:`allison.crt` and :file:`allison.key`.
+
+You should securely provide a copy of these files to the API end user, as well as a copy of the cluster's public certificate, the :file:`cluster.crt` file.
+
+Using an API Certificate to Authenticate
+========================================
+
+Once in possession of an API user certificate and the cluster certificate an end user must authenticate with those certificates in every request to the cluster REST API.
+The cluster certificate ensures the user is connecting to the genuine API of their cluster.
+The client certificate allows the API server to ensure the request is from a genuine, authorized user.
+An example of performing this authentication with ``cURL`` is given below.
+In this example, ``172.16.255.250`` represents the IP address of the control service.
+The following is an example of an authenticated request to create a new container on a cluster.
+
+On OS X
+-------
+
+Make sure you know the common name of the client certificate you will use.
+If you just generated the certificate following the :ref:`instructions above <generate-api>`, the common name is ``user-<username>`` where ``<username>`` is whatever argument you passed to ``flocker-ca generate-api-certificate``.
+If you're not sure what the username is, you can find the common name like this:
+
+.. code-block:: console
+
+    $ openssl x509 -in user.crt -noout -subject
+    subject= /OU=164b81dd-7e5d-4570-99c7-8baf1ffb49d3/CN=user-allison
+
+In this example, ``user-allison`` is the common name.
+Import the client certificate into the ``Keychain`` and then refer to it by its common name:
+
+.. code-block:: console
+
+    $ openssl pkcs12 -export -in user.crt -inkey user.key -out user.p12
+	Enter Export Password:
+	Verifying - Enter Export Password:
+    $ security import user.p12 -k ~/Library/Keychains/login.keychain
+    $ curl --cacert $PWD/cluster.crt --cert "<common name>" \
+         https://172.16.255.250:4523/v1/configuration/containers
+
+On Linux
+--------
+
+.. code-block:: console
+
+    $ curl --cacert $PWD/cluster.crt --cert $PWD/user.crt --key $PWD/user.key \
+         https://172.16.255.250:4523/v1/configuration/containers
+
+You can read more about how Flocker's authentication layer works in the :ref:`security and authentication guide <security>`.
+
+.. note::
+	Now you have set up an authenticated user you may want to perform the steps in :ref:`the MongoDB tutorial <movingapps>` to ensure that your nodes are correctly configured.
```

Comment:
I don't think the MongoDB tutorial can be run at this stage. This needs to move to the end of the top-level section (i.e. to Enabling the Flocker agent service).